File: SpringBoot-Backend/src/main/java/net/study/springboot/domain/Employee.java
package net.study.springboot.domain;

import jakarta.persistence.*;
import lombok.*;
import org.jetbrains.annotations.NotNull;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

import static java.lang.Character.isAlphabetic;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Builder
@Table(name="employees")
public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "empid")
    private int id;

    @NotNull
    @Column(name = "empfirstname")
    private String firstName;

    @NotNull
    @Column(name = "emplastname")
    private String lastName;

    @NotNull
    @Column(name = "empemail")
    private String emailId;

    @NotNull
    @Column(name = "empcity")
    private String city;

    @NotNull
    @Column(name = "empphone")
    private String phone; // corrected from Integer to String

    @Column(name = "empis_active")
    private boolean is_active;

    @Column(name = "empis_deleted")
    private boolean is_deleted;

    // ✅ Newly Added Fields

    @NotNull
    @Column(name = "empstate")
    private String state;

    @NotNull
    @Column(name = "empcountry")
    private String country;

    @NotNull
    @Column(name = "empdob")
    private LocalDate dob;

    @NotNull
    @Column(name = "empdate_join")
    private LocalDate dateOfJoining;

    @NotNull
    @Column(name = "empsalary")
    private BigDecimal salary;


    @Column(name = "emprating")
    private int rating;

    //other data fields you might want to add: DOB, start/end date, title or role, hourly wage
    @OneToMany(targetEntity = EmployeeHobbies.class, cascade = CascadeType.ALL)
    @JoinColumn(name="empid",referencedColumnName = "empid")
    private List<EmployeeHobbies> employeeHobbies;



}

File: Angular-Frontend/src/app/models/employee.ts
import { Hobby } from './hobby';
import { Department } from './department';

export interface EmployeeHobbyMap {
  id: number;
  employeeId: number;
  hobbyId: number;
  hobby: Hobby;
}

export interface EmployeeDepartmentMap {
  id: number;
  employeeId: number;
  departmentId: number;
  department: Department;
}

export class Employee { 
    id: number = 0;
    firstName: string = "";
    lastName: string = "";
    emailId: string = "";
    city: string = "";
    state: string = "";
    country: string = "";
    phone: string = ""; // Changed from number to string
    dob: string = ""; // Date of birth in YYYY-MM-DD format
    dateOfJoining: string = ""; // Date of joining in YYYY-MM-DD format
    salary: number = 0;
    rating: number = 1; // Rating from 1-10
    employeeHobbies: EmployeeHobbyMap[] = [];
    employeeDepartments: EmployeeDepartmentMap[] = [];
}
